snowshu/core/graph_set_runner.py

```diff
@@ -222,26 +224,49 @@ def _traverse_and_execute(self, executable: GraphExecutable) -> None:  # noqa: p
                         logger.info('Successfully extracted DDL statement for view '
                                     f'{executable.target_adapter.quoted_dot_notation(relation)}')
                     else:
-                        logger.info(
-                            f'Retrieving records from source {relation.temp_dot_notation}...')
+                        executable.source_adapter.create_table(
+                            query=relation.compiled_query,
+                            name=relation.name,
+                            schema=relation.temp_schema,
+                            database=relation.temp_database,
+                        )
+
                         try:
-                            executable.source_adapter.create_table(
-                                query=relation.compiled_query,
-                                name=relation.name,
-                                schema=relation.temp_schema,
-                                database=relation.temp_database,
+                            logger.info(
+                                f"Retrieving records from source {relation.temp_dot_notation}..."
                             )
                             fetch_query = f"SELECT * FROM {relation.temp_dot_notation}"
-                            query_data = executable.source_adapter.check_count_and_query(
-                                fetch_query, relation.sampling.max_allowed_rows, relation.unsampled)
+                            query_data = (
+                                executable.source_adapter.check_count_and_query(
+                                    fetch_query,
+                                    relation.sampling.max_allowed_rows,
+                                    relation.unsampled,
+                                )
+                            )
+                            relation.sample_size = len(query_data)
+                            logger.info(
+                                f"{relation.sample_size} records retrieved for relation {relation.dot_notation}."
+                            )
+                        # This except block is necessary due to VARIANT data type issues
+                        # in Snowflake. In the future, we should remove this and find a
+                        # better solution. 
```

Comment:
Temporary exception handling so the process doesn't fail but gives us information on where the issue happened. Issue is going to be investigated in separate ticket.